src/main/java/ch/wisv/areafiftylan/controller/MailRestController.java

```diff
@@ -0,0 +1,69 @@
+package ch.wisv.areafiftylan.controller;
+
+import ch.wisv.areafiftylan.dto.MailDTO;
+import ch.wisv.areafiftylan.exception.UserNotFoundException;
+import ch.wisv.areafiftylan.model.Team;
+import ch.wisv.areafiftylan.model.User;
+import ch.wisv.areafiftylan.service.MailService;
+import ch.wisv.areafiftylan.service.TeamService;
+import ch.wisv.areafiftylan.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.mail.MessagingException;
+
+@RestController
+@RequestMapping("/mail")
+public class MailRestController {
+
+    @Autowired
+    public MailRestController(MailService mailService, UserService userService, TeamService teamService) {
+        this.mailService = mailService;
+        this.userService = userService;
+        this.teamService = teamService;
+    }
+
+    MailService mailService;
+    UserService userService;
+    TeamService teamService;
+
+    @RequestMapping(value = "/user/{userId}", method = RequestMethod.POST)
+    ResponseEntity<?> sendMailToUser(@PathVariable Long userId, @Validated @RequestBody MailDTO mailDTO) {
+        User user = userService.getUserById(userId).orElseThrow(() -> new UserNotFoundException(userId));
+
+        try {
+            mailService.sendTemplateMailToUser(user, mailDTO);
+        } catch (MessagingException e) {
+            e.printStackTrace();
+        }
+
+        return new ResponseEntity<>(null, new HttpHeaders(), HttpStatus.OK);
+
+    }
+
+    @RequestMapping(value = "/team/{teamId}", method = RequestMethod.POST)
+    ResponseEntity<?> sendMailToTeam(@PathVariable Long teamId, @Validated @RequestBody MailDTO mailDTO) {
+        Team team = teamService.getTeamById(teamId);
+        try {
+            mailService.sendTemplateMailToTeam(team, mailDTO);
+        } catch (MessagingException e) {
+            e.printStackTrace();
+        }
```

Comment:
With this code, you get the team from the service, and then push it back to the service with the mail. A better manner will be to send the teamId and the mail directly to the service.